metaflow/client/core.py

```diff
@@ -290,23 +291,38 @@ def __init__(
         if pathspec:
             ids = pathspec.split("/")
 
+            if self._NAME == "flow" and len(ids) != 1:
+                raise MetaflowInvalidPathspec("Expects Flow('FlowName')")
+            elif self._NAME == "run" and len(ids) != 2:
+                raise MetaflowInvalidPathspec("Expects Run('FlowName/RunID')")
+            elif self._NAME == "step" and len(ids) != 3:
+                raise MetaflowInvalidPathspec("Expects Step('FlowName/RunID/StepName')")
+            elif self._NAME == "task" and len(ids) != 4:
+                raise MetaflowInvalidPathspec(
+                    "Expects Task('FlowName/RunID/StepName/TaskID')"
+                )
+            elif self._NAME == "artifact" and len(ids) != 5:
+                raise MetaflowInvalidPathspec(
+                    "Expects DataArtifact('FlowName/RunID/StepName/TaskID/ArtifactName')"
+                )
+
```

Comment:
Tried this out and seems to be working with the happy path. Original issue mentioned that pathspec should work with a trailing slash as well unless there's been a decision not to support this.

With the current implementation, a trailing `/` in the pathspec will raise a `MetaflowInvalidPathspec` even though the pathspec is otherwise valid

Reply:
we can punt adding support for trailing slash for now (to account for the scenario where we may want to attach a special meaning to trailing slash - highly unlikely)